packages/next/src/build/next-config-ts/loader.mts

```diff
@@ -0,0 +1,95 @@
+import type { LoadContext, ResolveContext } from './types'
+import { existsSync } from 'node:fs'
+import { extname } from 'node:path'
+import { isBareSpecifier, resolveSWCOptions } from './utils.js'
```

Comment:
why we use `.js` extension here? required by ts?

Reply:
Required by the output `loader.mjs`:

```shell
[Error: Cannot find module '/.../utils' imported from /.../loader.mjs] {
  url: 'file:///.../utils',
  code: 'ERR_MODULE_NOT_FOUND'
}
```

As [import Specifiers](https://nodejs.org/api/esm.html#import-specifiers) require the extensions when in relative:

> Relative specifiers like './startup.js' or '../config.mjs'. They refer to a path relative to the location of the importing file. **The file extension is always necessary for these**.